splink/missingness.py

```diff
@@ -0,0 +1,42 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import col, count, when
+
+from .charts import load_chart_definition, altair_if_installed_else_json
+
+import pandas as pd
+
+
+def missingness_chart(df: DataFrame):
+        """Produce bar chart of missingness in standardised nodes
+    Args:
+        df (DataFrame): Input Spark dataframe
+    Returns:
+        Bar chart of missingness
+        
+    """
+    
+    # Load JSON definition of missingness chart
+    chart_path = "missingness_chart_def.json"
+    missingness_chart_def = load_chart_definition(chart_path)
+    
+    
+    # Data for plot
+    # Count and percentage of nulls in each columns as pandas dataframe
+    df_nulls = df.select([count(when(col(c).isNull(), c)).alias(c) for c in df.columns])
+    pd_nulls = df_nulls.toPandas()
+    pd_nulls = pd.melt(pd_nulls)
+    
+    record_count = df.count()
+    pd_nulls["percentage"] = round(pd_nulls['value']*100/record_count, 1)
+    pd_nulls["percentage"] = pd_nulls['percentage'].astype(str) + "%"
+    
+    
+    # Add data to JSON chart definition
+    missingness_chart_def["data"]["values"] = pd_nulls.to_dict("records")
+    
+    # Update chart title
+    for c in missingness_chart_def["layer"]:
+        c["title"] = "Missingness per column out of {} records".format(record_count)
```

Comment:
Just saving you unnecessary hard coding of data formatting when you can leave it to the chart schema:

```suggestion
   record_count = df.count()
   pd_nulls["percentage"] = pd_nulls['value']/record_count

   # Add data to JSON chart definition
   missingness_chart_def["data"]["values"] = pd_nulls.to_dict("records")

   # Update chart title
   for c in missingness_chart_def["layer"]:
       c["title"] = f"Missingness per column out of {record_count:,.0f} records"
```

Reply:
Love the changes you've made, the new chart is beautiful! 😍